### ISSUE TITLE
BLAS test fails

### ISSUE THREAD

**Issue by loveshack** [flame/blis#177]
I tested the BLAS interface with one of the reference BLAS tests, and it aborted under dsymm with

libblis: Encountered invalid stride/dimension combination.

To reproduce, compile dblat3.f, linked against libblis (e.g. "gfortran -g -o dblat3 dblat3.f -l...", and run as

./dblat3 <dblat3.dat

The dblat3 files come from https://github.com/Reference-LAPACK/lapack/tree/master/BLAS/TESTING
If it doesn't abort, results are in dblat3.out.

BLIS should pass all the tests there, which are used for other BLAS implementations.

**Comment by fgvanzee** [comment]
Thanks for this report, Dave. I see the problem now and I'll begin working on a fix.

**Comment by fgvanzee** [comment]
@loveshack I think this issue is now addressed in c73055f0.

I was able to reproduce your issue pre-patch, and with the changes in c73055f0 the test driver ran to completion and exited normally. (I only tested `dblat3.f`, so there could be other problems with BLIS's BLAS layer, likely unrelated to this one.)

**Comment by loveshack** [comment]
It now gets through the BLAS tests that OpenBLAS runs (with an adjusted
makefile) but that doesn't examine the results of the error exit tests,
and those fail, e.g.

 ******* DGEMM  FAILED THE TESTS OF ERROR-EXITS *******

**Comment by fgvanzee** [comment]
@loveshack Could you clarify what you mean by error-exits? For example, is it a different `dblat` driver file that performs the error-exit tests? (Sorry, I don't have things set up to run all of them conveniently yet. I'm currently working on the `f2c` solution.)

**Comment by loveshack** [comment]
No, the testing is controlled by an input flag with the same driver
(line 7 in the example input for dblat3 below).  I haven't checked what
it does exactly, but it uses a replacement error-reporting routine
(xerbla?).  The stop on failures flag doesn't seem to include that.  (I
don't know how important it is to pass that.)

````
'DBLAT3.SUMM'     NAME OF SUMMARY OUTPUT FILE
6                 UNIT NUMBER OF SUMMARY FILE
'DBLAT3.SNAP'     NAME OF SNAPSHOT OUTPUT FILE
-1                UNIT NUMBER OF SNAPSHOT FILE (NOT USED IF .LT. 0)
F        LOGICAL FLAG, T TO REWIND SNAPSHOT FILE AFTER EACH RECORD.
T        LOGICAL FLAG, T TO STOP ON FAILURES.
T        LOGICAL FLAG, T TO TEST ERROR EXITS.
16.0     THRESHOLD VALUE OF TEST RATIO
7                 NUMBER OF VALUES OF N
0 1 2 3 7 31 63  VALUES OF N
3                 NUMBER OF VALUES OF ALPHA
0.0 1.0 0.7       VALUES OF ALPHA
3                 NUMBER OF VALUES OF BETA
0.0 1.0 1.3       VALUES OF BETA
DGEMM  T PUT F FOR NO TEST. SAME COLUMNS.
DSYMM  T PUT F FOR NO TEST. SAME COLUMNS.
DTRMM  F PUT F FOR NO TEST. SAME COLUMNS.
DTRSM  F PUT F FOR NO TEST. SAME COLUMNS.
DSYRK  F PUT F FOR NO TEST. SAME COLUMNS.
DSYR2K F PUT F FOR NO TEST. SAME COLUMNS.
````

**Comment by fgvanzee** [comment]
@loveshack I didn't notice that the driver was producing `dblat3.out`. I was erroneously assuming that no output to `stdout` meant no errors. Sorry about that. I understand what you mean by error-exit tests now:
```
 ******* XERBLA WAS CALLED WITH SRNAME = dgemm  INSTEAD OF DGEMM  *******
 ******* XERBLA WAS CALLED WITH SRNAME = dgemm  INSTEAD OF DGEMM  *******
 ******* XERBLA WAS CALLED WITH SRNAME = dgemm  INSTEAD OF DGEMM  *******
 ...
```
So it seems that the only reason the testsuite is complaining is because the name of the operation was provided in lowercase instead of uppercase. Seems rather nit-picky of the testsuite. Nonetheless, I've fixed the issue in cb7ed90.

**Comment by fgvanzee** [comment]
@loveshack I'm going to close this out now that we have the BLAS test suite available, which appears to pass everything (at least for me).
